src/NuGet.Clients/NuGet.VisualStudio.OnlineEnvironment.Client/SolutionExplorer/NuGetWorkspaceCommandHandler.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Threading;
+using Microsoft.VisualStudio.Workspace.VSIntegration.UI;
+
+namespace NuGet.VisualStudio.OnlineEnvironment.Client
+{
+    /// <summary>
+    /// Extends the Solution Explorer in online environment scenarios.
+    /// </summary>
+    internal class NuGetWorkspaceCommandHandler : IWorkspaceCommandHandler
+    {
+        private readonly JoinableTaskContext _taskContext;
+        private readonly IAsyncServiceProvider _serviceProvider;
+
+        private static bool IsNuGetFunctionalityAvailable = false;
+
+        public NuGetWorkspaceCommandHandler(JoinableTaskContext taskContext, IAsyncServiceProvider asyncServiceProvider)
+        {
+            _taskContext = taskContext;
+            _serviceProvider = asyncServiceProvider;
+        }
+
+        /// <summary>
+        /// The command handlers priority. If there are multiple handlers for a given node
+        /// then they are called in order of decreasing priority.
+        /// </summary>
+        public int Priority => 2000;
+
+        /// <summary>
+        /// Whether or not this handler should be ignored when multiple nodes are selected.
+        /// </summary>
+        public bool IgnoreOnMultiselect => true;
+
+        public int Exec(List<WorkspaceVisualNodeBase> selection, Guid pguidCmdGroup, uint nCmdID, uint nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
```

Comment:
If this method is invoked from outside of NuGet, does it need to be resilient to exceptions? Is it allowed as this method to throw?

Reply:
Think that's never null, but adjusted it to ensure NuGet does not throw because the docs don't explicitly say anything.